mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala

```diff
@@ -569,30 +577,43 @@ class DecisionTreeTest(sc: SparkContext) extends DecisionTreeTests(sc) {
     (splits, categoricalFeaturesInfo_, labelType)
   }
 
-  override def runTest(rdd: RDD[LabeledPoint]): RandomForestModel = {
+  override def runTest(rdd: RDD[LabeledPoint])
+    : Either[RandomForestModel, GradientBoostedTreesModel] = {
     val treeDepth: Int = intOptionValue(TREE_DEPTH)
     val maxBins: Int = intOptionValue(MAX_BINS)
     val numTrees: Int = intOptionValue(NUM_TREES)
     val featureSubsetStrategy: String = stringOptionValue(FEATURE_SUBSET_STRATEGY)
     val ensembleType: String = stringOptionValue(ENSEMBLE_TYPE)
-    if (!Array("RandomForest").contains(ensembleType)) {
+    if (!supportedTreeTypes.contains(ensembleType)) {
       throw new IllegalArgumentException(
         s"DecisionTreeTest given unknown ensembleType param: $ensembleType." +
-        " Supported values: RandomForest.")
+        " Supported values: " + supportedTreeTypes.mkString(" "))
     }
     if (labelType == 0) {
       // Regression
       ensembleType match {
         case "RandomForest" =>
-          RandomForest.trainRegressor(rdd, categoricalFeaturesInfo, numTrees, featureSubsetStrategy,
-            "variance", treeDepth, maxBins, this.getRandomSeed)
+          Left(RandomForest.trainRegressor(rdd, categoricalFeaturesInfo, numTrees, featureSubsetStrategy,
+            "variance", treeDepth, maxBins, this.getRandomSeed))
+        case "GradientBoostedTrees" =>
+          val treeStrategy = new Strategy(Algo.Regression, Variance, treeDepth,
+            labelType, maxBins, QuantileStrategy.Sort, categoricalFeaturesInfo)
+          val boostingStrategy = BoostingStrategy(treeStrategy, SquaredError, numIterations = 100,
+            learningRate = 0.1)
+          Right(GradientBoostedTrees.train(rdd, boostingStrategy))
       }
     } else if (labelType >= 2) {
       // Classification
       ensembleType match {
         case "RandomForest" =>
-          RandomForest.trainClassifier(rdd, labelType, categoricalFeaturesInfo, numTrees,
-            featureSubsetStrategy, "gini", treeDepth, maxBins, this.getRandomSeed)
+          Left(RandomForest.trainClassifier(rdd, labelType, categoricalFeaturesInfo, numTrees,
+            featureSubsetStrategy, "gini", treeDepth, maxBins, this.getRandomSeed))
+        case "GradientBoostedTrees" =>
+          val treeStrategy = new Strategy(Algo.Classification, Gini, treeDepth,
```

Comment:
Gini -> Variance  (This is ignored, but to be clear, GBTs use Variance for each tree since classification is rephrased as regression.)